File: cs/sample/Sample.Tests/ConvertBinaryLinkedList2Int/testConvertBinaryLinkedList2Int.cs
using System;
using Xunit;
using Sample.ConvertBinaryLinkedList2Int;
using Sample.ListNode;

namespace Sample.Tests{
    public class UnitTest_ConvertBinaryLinkedList2Int{
        [Theory]
        [InlineData("101",5)]
        [InlineData("0",0)]
        [InlineData("10",2)]
        [InlineData("100100111000000",18880)]
        public void checkResult(string s,int expected){
            listNode l = listNode.listNode_fromStr(s);
            Assert.Equal(expected,convertBinaryLinkedList2Int.naive(l));
        }
    }
}

File: py/tests/testConvertBinaryLinkedList2Int/test_ConvertBinaryLinkedList2Int.py
from solns.ListNode.ListNode import *
import unittest
from parameterized import parameterized as p
from solns.convertBinaryLinkedList2Int.convertBinaryLinkedList2Int import *

class UnitTest_ConvertBinaryLinkedList2Int(unittest.TestCase):
    @p.expand([
        ["101",5],["0",0],["10",2],["100100111000000",18880]
    ])
    def test_naive(self,s,expected):
        l = ListNode.listNode_fromStr(s)
        self.assertEqual(Solution.naive(l),expected)
